opencti-platform/opencti-graphql/src/database/redis.ts

```diff
@@ -189,66 +197,109 @@ export const redisTx = async (client: Cluster | Redis, chain: (tx: ChainableComm
+      const sessions = await redisTx(getClientBase(), async (tx) => {
+        for (let index = 0; index < partInstances.length; index += 1) {
+          const key = partInstances[index];
+          tx.get(key).ttl(key);
+          if (new Date().getTime() - startProcessingTime > MAX_EVENT_LOOP_PROCESSING_TIME) {
+            startProcessingTime = new Date().getTime();
+            await new Promise((resolve) => {
+              setImmediate(resolve);
+            });
+          }
         }
-        const [, ttl] = results[0];
-        const [, data] = results[1] as string[];
-        return data ? { id: i, ttl, data } : null;
-      })));
-    return instancesConfig.filter(filterEmpty).map((n) => {
-      return { redis_key_id: n.id, redis_key_ttl: n.ttl, ...JSON.parse(n.data) };
-    });
+      }) ?? [];
+      startProcessingTime = new Date().getTime();
+      for (let n = 0; n < sessions.length; n += 2) {
+        const cookie = sessions[n][1] as string;
+        if (cookie) {
+          const ttl = sessions[n + 1][1];
+          const id = partInstances[n / 2];
+          keysElements.push({ redis_key_id: id, redis_key_ttl: ttl, ...JSON.parse(cookie) });
+        }
+        if (new Date().getTime() - startProcessingTime > MAX_EVENT_LOOP_PROCESSING_TIME) {
+          startProcessingTime = new Date().getTime();
+          await new Promise((resolve) => {
+            setImmediate(resolve);
+          });
+        }
+      }
+    }
+    return keysElements;
   }
   return [];
 };
 // endregion
 
 // region session
+const PLATFORM_KEY_SESSIONS = 'platform_sessions';
 export const clearSessions = async () => {
-  const contextIds = await getClientBase().zrange('platform_sessions', 0, -1);
-  return Promise.all(contextIds.map((id) => getClientBase().del(id)));
+  const instanceKeys = await getClientBase().zrange(PLATFORM_KEY_SESSIONS, 0, -1);
+  if (instanceKeys && instanceKeys.length > 0) {
+    const keysBulk = R.splitEvery(MAX_MULTI_REDIS_OPERATIONS, instanceKeys);
+    for (let ind = 0; ind < keysBulk.length; ind += 1) {
+      const partInstances = keysBulk[ind];
+      await redisTx(getClientBase(), async (tx) => {
+        let startProcessingTime = new Date().getTime();
+        for (let index = 0; index < partInstances.length; index += 1) {
+          const key = partInstances[index];
+          tx.del(key);
+          if (new Date().getTime() - startProcessingTime > MAX_EVENT_LOOP_PROCESSING_TIME) {
+            startProcessingTime = new Date().getTime();
+            await new Promise((resolve) => {
+              setImmediate(resolve);
+            });
+          }
+        }
+      });
+    }
+  }
+  await getClientBase().del(PLATFORM_KEY_SESSIONS);
+  return true;
 };
-export const getSession = async (key: string) => {
+export const getSession = async (baseKeyId: string) => {
+  const key = `{${PLATFORM_KEY_SESSIONS}}:${baseKeyId}`; // Key of a list must be attached to the same redis slot
```

Comment:
I don't understand, does it mean the session key has now PLATFORM_KEY_SESSIONS as a prefix ?

Reply:
something is wrong with this `{${PLATFORM_KEY_SESSIONS}}:${baseKeyId}`, when I debug this method, I find myself with a duplicate {${PLATFORM_KEY_SESSIONS}}: prefix, I'm not sure we should add it here, since the key is already named in redis with this prefix